servers/rendering/shader_compiler.cpp

```diff
@@ -1107,7 +1107,9 @@ String ShaderCompiler::_dump_node_code(const SL::Node *p_node, int p_level, Gene
 					} else {
 						code += "";
 					}
-					code += _dump_node_code(cnode->array_declarations[0].initializer[i], p_level, r_gen_code, p_actions, p_default_actions, p_assigning);
+					if (i < cnode->array_declarations[0].initializer.size()) {
+						code += _dump_node_code(cnode->array_declarations[0].initializer[i], p_level, r_gen_code, p_actions, p_default_actions, p_assigning);
+					}
```

Comment:
To be honest, I would leave as it is for now in this PR, and fix it in other places where initializer used in separate PR. Also, maybe this need to be wrapped with `ERR_FAIL_COND` eg.  `if (ERR_FAIL_COND(i >= cnode->array_declarations[0].initializer.size())`